src/NHibernate/Param/NamedParameter.cs

```diff
@@ -15,6 +15,8 @@ public NamedParameter(string name, object value, IType type)
 		public object Value { get; internal set; }
 		public IType Type { get; internal set; }
 
+		public virtual bool IsCollection => false;
```

Comment:
Probably just create a constructor to set this value instead of a new type.

Reply:
Done.